### ISSUE TITLE
Automated packaging and upload of WMCore services to Pypi

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11276]
**Impact of the new feature**
WMCore in general

**Is your feature request related to a problem? Please describe.**
As part of the migration away from RPMs, we need to start building WMCore services in a more standardized and simple way, and PyPi is a great choice for packaging and distribution of Python-based software.

**Describe the solution you'd like**
Automate the process of packaging and uploading new versions of WMCore services to PyPi. 
Expanding on the details, I think we should start with an automatic build/upload to Pypi whenever a new WMCore release is made (not a tag, but the "final" release). We need create a package for each of the services provided in WMCore, a non-exhaustive list is:
* global workqueue
* reqmgr2
* reqmon
* t0-reqmon
* wmagent
* reqmgr2ms (or each of those microservices in their own package).

This is likely to be implemented with GitHub actions, but if we decide that Jenkins would fit better our project, that is Ok as well.

**Describe alternatives you've considered**
There are many GH issues associated to this, under the "containerization" label:
https://github.com/dmwm/WMCore/issues?q=is%3Aopen+is%3Aissue+label%3Acontainerization

**Additional context**
Note that this issue is only addressing the automated packaging and distribution. Deploying our WMCore services from PyPi should be addressed in a new GH issue.

**Comment by todor-ivanov** [comment]
Thanks @amaltaro for summarizing this in this well constructed  issue. I'd like to have just one comment here:
> we should start with an automatic build/upload to Pypi whenever a new WMCore release is made (not a tag, but the "final" release)

I do agree that in the production Pypi repository (i.e. pypi.org) needs to go only the final release. But, we should create some github actions which would build and upload the pre-releases to the test Pypi repository test.pypi.org - this is  a completely independent one. These actions can be triggered based on tag names suffixes like `v2.1.2pre*` etc.  And I think this is a must, so that the whole validation process can go through the pypi deployment procedures, rather than using the rpm based builds to validate the pre-release and then simply upload the final release packages to pypi. This alone does not guarantee that starting with a pure python pakcaged deployment one would achieve the same result as the one deployed through the rpms in the production system. 

In summary my suggestion goes along the line of starting to think from the very beginning, to not only build and porvide some packages through pypi, but rather to start projecting for fully shifting the whole process way from the rpm based packaging system, together with the release validation procedures.

**Comment by khurtado** [comment]
@amaltaro  #11318 takes care of the following packages: `[wmagent, wmagent-devtools, wmcore, reqmon, reqmgr2, reqmgr2ms, global-workqueue, acdcserver]`

Notes:
- t0-reqmon is the same as reqmon (when it comes to building the python package), so we are not building this
Ref:
https://github.com/cms-sw/cmsdist/blob/comp_gcc630/t0_reqmon.spec
https://github.com/cms-sw/cmsdist/blob/comp_gcc630/reqmon.spec
- t0 spec relies on 2 github repositories with packages that apparently have the same name. They are already uploading a t0 python package on pypy here: https://pypi.org/project/t0/
https://github.com/cms-sw/cmsdist/blob/comp_gcc630/t0.spec
I'm not sure if we need to rename our package component or not, but this should be a different GH issue
